public_desktop/desktop-app.js

```diff
@@ -90,8 +92,60 @@ function startDesktopApp() {
 		}
 	}
 
+	function contextMenu( ev ) {
+		var menu = {};
+		var showEditorMenu = false;
+
+		// check if in visual editor, ipc sends an EventEmitter
+		if ( typeof ev.sender !== 'undefined' )  {
+			showEditorMenu = true;
+		}
+		// check if in textarea or similar
+		else if ( ( typeof ev.target !== 'undefined' ) && ev.target.closest( 'textarea, input, [contenteditable="true"]' ) ) {
+			showEditorMenu = true;
+		}
+
+		if ( showEditorMenu ) {
+			menu = buildEditorContextMenu(selection);
+		}
+		else {
+			var selectedText = window.getSelection().toString();
+			menu = buildGeneralContextMenu(selectedText);
+		}
+
+		// The 'contextmenu' event is emitted after 'selectionchange' has fired but possibly before the
+		// visible selection has changed. Try to wait to show the menu until after that, otherwise the
+		// visible selection will update after the menu dismisses and look weird.
+		setTimeout(function() {
+			menu.popup(electron.remote.getCurrentWindow());
+		}, 30);
+	}
+
+	function resetSelection() {
+		selection = {
+			isMisspelled: false,
+			spellingSuggestions: []
+		};
+	}
+
+
 	debug = gGebug( 'desktop:browser' );
 
+
+	debug( 'Setting up Context Menus' );
+	try {
+		var buildEditorContextMenu = electron.remote.require( '../desktop/lib/menu/editor-context-menu' );
+		var buildGeneralContextMenu = electron.remote.require( '../desktop/lib/menu/general-context-menu' );
```

Comment:
That is based upon this setting of desktop global variable, not related to a module
https://github.com/Automattic/wp-desktop/blob/master/desktop/env.js#L84

It could be a solution to load the context menus into a global and use, but should be able to use the `remote.require` method to do so. Just doesn't want to work on Windows 😕